### PR TITLE
add arithmetic corner cases tests

### PR DESCRIPTION
**Pull request by oclaw**

### CODE REVIEW

File: proof-producer/tests/bin/proof-producer/resources/traces/contracts/tracer_data.sol [hunk 1]
```diff
@@ -25,3 +25,45 @@ contract SimpleStorage {
         emit ValueChanged(value);
     }
 }
+
+contract Uint256CornerCaseTests {
+    /// @notice Adds two uint256 numbers using EVM ADD opcode.
+    ///         Overflow results in 256-bit wrap-around, no revert.
+    function addAsm(uint256 a, uint256 b) external pure returns (uint256 result) {
+        assembly {
```
Comment (oclaw) [inline proof-producer/tests/bin/proof-producer/resources/traces/contracts/tracer_data.sol:33]: Replacement with direct arithm operator may result in solc compiler placing revert in case of overflow

**Review by akokoshn — APPROVED** [review]

**Review by akokoshn — APPROVED** [review]
